src/pymmcore_widgets/hcs/_calibration_widget/_calibration_sub_widgets.py

```diff
@@ -0,0 +1,382 @@
+from __future__ import annotations
+
+import string
+from pathlib import Path
+from typing import (
+    Any,
+    Iterable,
+    NamedTuple,
+    Union,
+    cast,
+)
+
+from fonticon_mdi6 import MDI6
+from pymmcore_plus import CMMCorePlus
+from qtpy.QtCore import QSize, Qt, Signal
+from qtpy.QtGui import QIcon, QPixmap
+from qtpy.QtWidgets import (
+    QComboBox,
+    QGroupBox,
+    QHBoxLayout,
+    QLabel,
+    QPushButton,
+    QSizePolicy,
+    QWidget,
+)
+from superqt.fonticon import icon
+from superqt.utils import signals_blocked
+from useq import WellPlate  # noqa: TCH002
+
+from pymmcore_widgets.hcs._graphics_items import RED, Well
+from pymmcore_widgets.useq_widgets._column_info import FloatColumn
+from pymmcore_widgets.useq_widgets._data_table import DataTableWidget
+
+AlignCenter = Qt.AlignmentFlag.AlignCenter
+FixedSizePolicy = (QSizePolicy.Policy.Fixed, QSizePolicy.Policy.Fixed)
+
+ALPHABET = string.ascii_uppercase
+ROLE = Qt.ItemDataRole.UserRole + 1
+
+MAX = 9999999
+
+ICON_PATH = Path(__file__).parent / "icons"
+ICON_SIZE = 22
+
+CIRCLE_ICON = QIcon(str(ICON_PATH / "circle-center.svg"))
+CIRCLE_TEXT = "1 points : add 1 points at the center of the well"
+CIRCLE_CENTER_POINTS = 1
+
+CIRCLE_EDGES_ICON = QIcon(str(ICON_PATH / "circle-edges.svg"))
+CIRCLE_EDGES_TEXT = "3 points : add 3 points on the circonference of the well"
+CIRCLE_EDGES_POINTS = 3
+
+SIDES_ICON = QIcon(str(ICON_PATH / "square-edges.svg"))
+SIDES_ITEM = "4 points: add 4 points, 1 per side of the rectangular/square well"
+SIDES_POINTS = 4
+
+VERTICES_ICON = QIcon(str(ICON_PATH / "square-vertices.svg"))
+VERTICES_TEXT = (
+    "2 points: add 2 points at 2 opposite vertices of the rectangular/square well"
+)
+VERTICES_POINTS = 2
+
+LABEL_STYLE = """
+    background: #00FF00;
+    font-size: 16pt; font-weight:bold;
+    color : black;
+    border: 1px solid black;
+    border-radius: 5px;
+"""
+
+
+class OnePoint(NamedTuple):
+    """One center point to calibrate a circular well."""
+
+    icon: QIcon = CIRCLE_ICON
+    text: str = CIRCLE_TEXT
+    points: int = CIRCLE_CENTER_POINTS
+
+
+class TwoPoints(NamedTuple):
+    """Two vertices points to calibrate a rectangular/square well."""
+
+    icon: QIcon = VERTICES_ICON
+    text: str = VERTICES_TEXT
+    points: int = VERTICES_POINTS
+
+
+class ThreePoints(NamedTuple):
+    """Three edge points to calibrate a circular well."""
+
+    icon: QIcon = CIRCLE_EDGES_ICON
+    text: str = CIRCLE_EDGES_TEXT
+    points: int = CIRCLE_EDGES_POINTS
+
+
+class FourPoints(NamedTuple):
+    """Four edge points to calibrate a rectangular/square well."""
+
+    icon: QIcon = SIDES_ICON
+    text: str = SIDES_ITEM
+    points: int = SIDES_POINTS
```

Comment:
these classes look more like single instances... you don't ever instantiate them with anything other than their defaults. 

```python
class Mode(NamedTuple):
    """Calibration mode."""

    icon: QIcon
    text: str
    points: int

OnePoint = Mode(CIRCLE_ICON, CIRCLE_TEXT, CIRCLE_CENTER_POINTS)
TwoPoints = Mode(VERTICES_ICON, VERTICES_TEXT, VERTICES_POINTS)
ThreePoints = Mode(CIRCLE_EDGES_ICON, CIRCLE_EDGES_TEXT, CIRCLE_EDGES_POINTS)
FourPoints = Mode(SIDES_ICON, SIDES_ITEM, SIDES_POINTS)
```

and then elsewhere, `modes: list[Mode]`

Reply:
and at that rate... since all the module-level variables are only used once... get rid of the module level variables and use literals when constructing each mode:

```python
OnePoint = Mode(
    QIcon(str(ICON_PATH / "circle-center.svg")),
    "1 points : add 1 points at the center of the well",
    1,
)
```

and when you need them in tests, use, eg., `OnePoint.points`